samples/algorithms/integer-factorization/FlameGraphCounter.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators;
+
+namespace Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime {
+    public class FlameGraphCounter : IQCTraceSimulatorListener {
+        private readonly PrimitiveOperationsCounterConfiguration configuration;
+        
+        /// <summary>
+        /// Contains all fully specified calls including resource count for the operation excluding all children.
+        /// </summary>
+        public Dictionary<string, double> FlameGraphData { get; } = new Dictionary<string, double>();
+
+        /// <summary>
+        /// Sets which resource to track (e.g., CNOT)
```

Comment:
Is this the only resource that shows up in the flamegraph, then?

Reply:
Yes